types/formdata.d.ts

```diff
@@ -47,51 +45,60 @@ export declare class FormData {
    *
    * @returns A `FormDataEntryValue` containing the value. If the key doesn't exist, the method returns null.
    */
-  readonly get: (name: string) => FormDataEntryValue | null
+  get(name: string): FormDataEntryValue | null
+
   /**
    * Returns all the values associated with a given key from within a `FormData` object.
    *
    * @param {string} name A name of the value you want to retrieve.
    *
    * @returns An array of `FormDataEntryValue` whose key matches the value passed in the `name` parameter. If the key doesn't exist, the method returns an empty list.
    */
-  readonly getAll: (name: string) => FormDataEntryValue[]
+  getAll(name: string): FormDataEntryValue[]
+
   /**
    * Returns a boolean stating whether a `FormData` object contains a certain key.
    *
    * @param name A string representing the name of the key you want to test for.
    *
    * @return A boolean value.
    */
-  readonly has: (name: string) => boolean
+  has(name: string): boolean
+
   /**
    * Deletes a key and its value(s) from a `FormData` object.
    *
    * @param name The name of the key you want to delete.
    */
-  readonly delete: (name: string) => void
+  delete(name: string): void
+
   /**
    * Returns an [`iterator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) allowing to go through all keys contained in this `FormData` object.
    * Each key is a `string`.
    */
-  readonly keys: () => Generator<string>
+  keys(): Generator<string>
+
   /**
    * Returns an [`iterator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) allowing to go through the `FormData` key/value pairs.
    * The key of each pair is a string; the value is a [`FormDataValue`](https://developer.mozilla.org/en-US/docs/Web/API/FormDataEntryValue).
    */
-  readonly entries: () => Generator<[string, FormDataEntryValue]>
+  entries(): Generator<[string, FormDataEntryValue]>
+
   /**
    * Returns an [`iterator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) allowing to go through all values contained in this object `FormData` object.
    * Each value is a [`FormDataValue`](https://developer.mozilla.org/en-US/docs/Web/API/FormDataEntryValue).
    */
-  readonly values: () => Generator<FormDataEntryValue>
+  values(): Generator<FormDataEntryValue>
+
   /**
    * An alias for FormData#entries()
    */
-  readonly [Symbol.iterator]: () => Generator<[string, FormDataEntryValue], any, unknown>
+  [Symbol.iterator](): Generator<[string, FormDataEntryValue], void>
+
   /**
    * Executes given callback function for each field of the FormData instance
    */
-  readonly forEach: (fn: (value: FormDataEntryValue, key: string, fd: FormData) => void, ctx?: unknown) => void
-  get [Symbol.toStringTag] (): string
+  forEach(fn: (value: FormDataEntryValue, key: string, fd: FormData) => void, ctx?: unknown): void
```

Comment:
`fd` -> `formData`

Reply:
> It would be good to rename fn to something more descriptive as well but function is a reserved keyword

Maybe to `callback`?

Reply:
Sounds good. This is also how MDN describes `forEach`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach (for Map though, but still relevant)

Reply:
See: https://github.com/microsoft/TypeScript/blob/5546f207c5d20578f1d215ccc110dd42bf8d359b/lib/lib.dom.d.ts#L5413

Reply:
Can we use those interfaces instead? Or should we not?

Reply:
Not so sure about that. At least `@types/node` has separate declarations for `Blob`.

Reply:
> Sounds good. This is also how MDN describes forEach: developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach (for Map though, but still relevant)

Can't find documentation for this method in FormData, though it exists in browsers. The only place it appears is compatibility table on this page: https://developer.mozilla.org/en-US/docs/Web/API/FormData#browser_compatibility
Even the spec does not say anything about this method, where FormData is described: https://xhr.spec.whatwg.org/#interface-formdata

Reply:
> Can we use those interfaces instead? Or should we not?

By the way, at least for FormData constructor Node and Browser versions aren't compatible: because browser version takes `form` argument, which is DOM element. See: https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData#parameters

Reply:
https://github.com/nodejs/undici/blob/b0ec94d1a4b679b212e441ef332eb1337ad92dbd/lib/fetch/formdata.js#L9-L17

https://github.com/nodejs/undici/blob/b0ec94d1a4b679b212e441ef332eb1337ad92dbd/lib/fetch/util.js#L287

There might be a use case where [`jsdom`](https://github.com/jsdom/jsdom) can be used 🤔

Reply:
@ronag Should we later change the error message to e.g.

```
Failed to constructor 'FormData': parameter 'form' is unsupported
```